src/routers/user.ts

```diff
@@ -21,7 +21,7 @@ user.post('/user', userBodyValidatorOnCreate, async (req, res) => {
   }
 });
 
-user.get('/user/:id', userParamsIdValidator, async (req, res) => {
+user.get('/user/:id', paramsIdValidator, async (req, res) => {
```

Comment:
You validated body (and params and query in methods below), so you pass not just request, but validated request. Please, add specific type for your request (look up `express-joi-validation` lib for examples). They will also spare you from some unnecessary type assertions.

Reply:
fixed